perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/GroupsManagerBlImpl.java

```diff
@@ -2951,11 +2946,8 @@ private void categorizeMembersForLightweightSynchronization(PerunSession sess, G
 						}
 				}
 				idsOfUsersInGroup.remove(user.getId());
-			} else if (candidate != null) {
-				candidatesToAdd.add(candidate);
 			} else {
-				//Both null means that we can't find subject by login in extSource at all (will be in skipped members)
-				log.debug("Subject with login {} was skipped because can't be found in extSource {}.", login, memberSource);
+				candidatesToAdd.add(candidate);
 			}
```

Comment:
This is now much harder to read so we probably need to change the code a little.
I would suggest defining all possible situations we can reach:

1. The user was found by extSource or additional user extSource => user is not null, need to process more
2. The user wasn't found => user is null, the situation will be logged and skipped

And you can say that there are only 2 possible ways how to process the situation 1:
1. the user is already a member of the group =>just revalidate him if possible and remove him from the list idsOfUsersInGroup
2. the user is not yet a member of the group => create a candidate, add him to candidatesToAdd

What do you think? In my opinion, this way it will be much easier for reading.

Reply:
I agree that the current state is difficult to understand, the suggested change looks good.

Reply:
I hope it's better now.